bridges/snowbridge/pallets/ethereum-client/src/lib.rs

```diff
@@ -465,11 +470,17 @@ pub mod pallet {
 				});
 			};
 
+			let pays_fee = Self::check_refundable(update, latest_finalized_state.slot);
+			let actual_weight = match update.next_sync_committee_update {
+				None => T::WeightInfo::submit(),
+				Some(_) => T::WeightInfo::submit_with_sync_committee(),
+			};
+
 			if update.finalized_header.slot > latest_finalized_state.slot {
 				Self::store_finalized_header(update.finalized_header, update.block_roots_root)?;
 			}
 
-			Ok(())
+			Ok(PostDispatchInfo { actual_weight: Some(actual_weight), pays_fee })
```

Comment:
Can you double-check and confirm that this doesn't allow free spamming for this call with using the same "valid" update (with `update.next_sync_committee_update.is_some() == true`) over and over for free, potentially DoS-ing BH blocks?

It looks like there is [a `compute_period` verification](https://github.com/paritytech/polkadot-sdk/blob/6ad199baa81035bbeb3f3859c1ad730f08b1a04a/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L313C1-L322C5) for updating sync committee, that would not allow the DoS I described above, but I got a bit lost in the details and would like you to confirm.

Reply:
@acatangiu that's a valid point, thanks! There is a specific test for that (`duplicate_sync_committee_updates_are_not_free`): https://github.com/paritytech/polkadot-sdk/pull/5201/files#diff-ea6025b3cac6719f750d1f0065ba30d70a16d386a5ce374e4a6b79299f48a3eeR684

Reply:
Thanks, didn't yet look at tests 🙈

Reply:
The test tries to submit the exact same proof twice, which will fail the second time because the attested header in the proof is no longer newer than the stored one on-chain.

But what about the following scenario:
- Relayer submits valid update for slot `N` in period `P` with sync committee `S` provided in `Some(next_sync_committee_update)`
- Relayer then provides update for slot `N+1` in period `P` with sync committee `S` provided in `Some(next_sync_committee_update)`?

Basically try to "farm" the `Some(next_sync_committee_update)` tx fee freebie, to spam update one slot at a time.

Reply:
The above is a valid issue, and is being fixed in https://github.com/Snowfork/polkadot-sdk/pull/172